### PR TITLE
Add local map server resources

### PR DESCRIPTION
**Pull request by junqiu-lei**
Signed-off-by: Junqiu Lei <junqiu@amazon.com>

### Description
Add local map server docker resources that help OpenSearch user to self-host maps server when they are not able to access to OpenSearch default maps services.

More details about this project info, please look at the README.MD file.

### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: local-maps-server/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (navneet1v) [inline local-maps-server/.gitignore:1]: don't we need `npm-debug.log` in the git ignore too?

File: local-maps-server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM node:16
+
+WORKDIR /usr/src/app
+
+COPY package*.json ./
+RUN npm install
+
+COPY . .
```
Comment (navneet1v) [inline local-maps-server/Dockerfile:8]: what is this used for?

File: local-maps-server/README.MD [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# OpenSearch Local Map Server
+
+This project helps to create a self-host maps server for OpenSearch Dashboard, for these who are not able to access to OpenSearch default maps service on air-gapped environments.
+
+The server provides maps manifest url(including both maps tiles and vectors), maps tiles url and maps vectors url which are compatible with OpenSearch.
+
+## Build docker image
+```
+docker build -t osm-server .
+```
```
Comment (navneet1v) [inline local-maps-server/README.MD:10]: should we provide the docker installation guidelines?

File: local-maps-server/app.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+const cors = require('cors');
+const app = express()
+const port = 8080
+
+app.use(cors());
+
+app.use(express.static('public'));
+
+app.listen(port, () => {
+    console.log('Maps server start running')
```
Comment (navneet1v) [inline local-maps-server/app.js:11]: console.log is not recommended for production cases.

File: local-maps-server/app.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const express = require('express')
+const cors = require('cors');
+const app = express()
+const port = 8080
+
```
Comment (navneet1v) [inline local-maps-server/app.js:5]: why we are using .js code why not use the TypeScript?

File: local-maps-server/public/index.html [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+<script>
+    var map = new maplibregl.Map({
+        container: 'map',
+        style: {
+            'version': 8,
+            'sources': {
+                'raster-tiles': {
+                    'type': 'raster',
+                    'tiles': [
+                        'http://localhost:8080/tiles/data/{z}/{x}/{y}.png'
```
Comment (navneet1v) [inline local-maps-server/public/index.html:25]: localhost?

File: local-maps-server/package.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (navneet1v) [inline local-maps-server/package.json:1]: what about other commands like npm dev etc?

File: local-maps-server/public/index.html (hunk 6 above)
Comment (junqiu-lei) [reply local-maps-server/public/index.html:25]: Thanks! Will update to pass the host machine ip to nodejs env and then create the static json file correspondingly.

File: local-maps-server/services/osm.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+import { hostUrl } from "../common";
+
+export const osm = {
+  tilejson: "2.0.0",
+  name: "OSM",
+  attribution:
+    '<a href="https://www.openstreetmap.org/copyright" target="_blank">&copy; OpenStreetMap contributors</a>',
+  minzoom: 0,
+  maxzoom: 22,
```
Comment (VijayanB) [inline local-maps-server/services/osm.ts:14]: what happens if i have only zoom up to 8?, will that be a problem?

Comment (junqiu-lei) [reply local-maps-server/services/osm.ts:14]: So when set maximum zoom to 22 here, if import tiles set is zoom 0-8, when users click into zoom 9, the tile map will be empty, and they can click zoom back to 8. The reason I used a maximum 22 is that users won't be limited whatever zoom level tiles set they imported.

Also we can provide a default value here and users can define the maximum zoom value by a docker env var when they run the server container.

Comment (VijayanB) [reply local-maps-server/services/osm.ts:14]: Got it.

**Review by VijayanB — DISMISSED** [review]

**Review by VijayanB — APPROVED** [review]
LGTM. Thanks

**Review by navneet1v — APPROVED** [review]

**Pull request merged**